be/src/storage/tablet_updates.cpp

```diff
@@ -1028,9 +1049,21 @@ void TabletUpdates::_wait_apply_done() {
     }
 }
 
-void TabletUpdates::_stop_and_wait_apply_done() {
+void TabletUpdates::stop_and_wait_apply_done() {
+    int64_t start_time = MonotonicMicros();
     _apply_stopped = true;
     _wait_apply_done();
+    int64_t duration = MonotonicMicros() - start_time;
+    StarRocksMetrics::instance()->primary_key_wait_apply_done_duration_ms.increment(duration / 1000);
```

Comment:
this is a gauge metrics, the `increment()` accumulates the time cost, but the number of _wait_apply_done() is not recorded, so the average latency can't be told from this single metrics.